[dlpack][compat] Add compatibility change for some DLPack usecases

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22691 [dlpack][compat] Add compatibility change for some DLPack usecases**

Differential Revision: [D16208381](https://our.internmc.facebook.com/intern/diff/D16208381)

bwasti committed:
[dlpack][compat] Add compatibility change for some DLPack usecases

wanchaol commented on aten/src/ATen/DLConvertor.cpp:
```diff
@@ -154,6 +154,8 @@ struct ATenDLMTensor {
 };
 
 void deleter(DLManagedTensor* arg) {
+  arg->manager_ctx = nullptr;
```
if you set `arg->manager_ctx` to `nullptr`, `delete static_cast<ATenDLMTensor*>(arg->manager_ctx)` won't work as before?

yinghai reviewed (commented):
Where are the places that we use `DLManagedTensor` instead of `DLTensor`?

bwasti committed:
Update on "[dlpack][compat] Add compatibility change for some DLPack usecases"

[dlpack][compat] Add compatibility change for some DLPack usecases

gh-metadata: pytorch pytorch 22691 gh/bwasti/26/head

bwasti committed:
Update on "[dlpack][compat] Add compatibility change for some DLPack usecases"

[dlpack][compat] Add compatibility change for some DLPack usecases

gh-metadata: pytorch pytorch 22691 gh/bwasti/26/head
